monitoring/rid_qualifier/flight_track_factory.py

```diff
@@ -0,0 +1,178 @@
+from shapely.geometry import Point, Polygon
+import shapely.geometry
+from pyproj import Geod, Transformer
+import json
+from pathlib import Path
+from typing import List, ClassVar
+
+class AdjacentCircularFlightPathsGenerator():
+
+    ''' A class to generate Flight Paths given a bounding box, this is the main module to generate flight path datasets, the data is generated as latitude / longitude pairs with assoiated with the flights. Additional flight metadata e.g. flight id, altitude, registration number can also be generated '''
+
+
+    flight_points: ClassVar[List] = []   # This is a object that containts multiple lists of flight tracks as points, in the latitude, longitude, altitude in tuple format. Depending on how the grid is generated in this case 3 columns and 2 rows with six flight tracks there will be six lists in this object
+    flight_grid: ClassVar[List] = [] # This object holds the polygon objects for the different grid cells within the bounding box. 
+    query_bboxes: ClassVar[List] = [] # This object holds the name and the polygon object of the query boxes. The number of bboxes are controlled by the `box_diagonals` variable
+
+    def __init__(self, minx: float, miny: float, maxx: float, maxy: float) -> None: 
+        # A bounding box (extents) is necessary, here the minx and miny detail the SE corner where minx is the longitude and miny is the latitude and maxx and maxy is the NW corner and maxx is the longitude and maxy is the latitude. Once these extents are specified, a grid needs to be created with two rows. The max_cols parameter specifies the desired columns inside the extents. The idea is that multiple flights will be created within the extents
```

Comment:
Where possible, let's try to follow something equivalent to [PEP257](https://www.python.org/dev/peps/pep-0257/#id17) -- in this case, I recommend something like:

```python
  def __init__(self, minx: float, miny: float, maxx: float, maxy: float) -> None:
    """Create an AdjacentCircularFlightPathsGenerator with the specified bounding box.

    Once these extents are specified, a grid will be created with two rows.  The idea is that multiple
    flights will be created within the extents.
    Args:
      minx: Western edge of bounding box (degrees longitude)
      maxx: Eastern edge of bounding box (degrees longitude)
      miny: Southern edge of bounding box (degrees latitude)
      maxy: Northern edge of bounding box (degrees latitude

    Raises:
      ValueError: If bounding box has more area than a 500m x 500m square.
    """
```

The `max_cols` text is removed because there is no `max_cols` argument any more.

Reply:
Yes, Thanks for this, I missed removing max_cols sentence, I included your suggestion as is.